src/librustc_trans/back/symbol_export.rs

```diff
@@ -228,7 +228,5 @@ fn symbol_for_def_id<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 
     let instance = Instance::mono(tcx, def_id);
 
-    symbol_map.get(TransItem::Fn(instance))
-              .map(str::to_owned)
-              .unwrap_or_else(|| symbol_name(instance, tcx))
+    str::to_owned(&symbol_name(instance, tcx))
```

Comment:
Shouldn't we just replace this entire function with a call to `symbol_name` (and change return type to `InternedString`, of course)? It seems like we should remove the secondary caching performed by `SymbolMap`, and instead have that also use the queries, no?